lib/recurly/Http.js

```diff
@@ -1,24 +1,64 @@
+const https = require('https')
+const casters = require('./Caster')
+
+// an adapter function to abstract away some of
+// the low level details of requests and responses in node
+function makeRequest (options, requestBody) {
+  if (requestBody) {
+    requestBody = JSON.stringify(casters.castRequest(requestBody))
+    options.headers['Content-Length'] = Buffer.byteLength(requestBody)
+  }
+  return new Promise((resolve, reject) => {
+    const httpRequest = https.request(options, (httpResponse) => {
+      const rbody = []
+      httpResponse.setEncoding('utf8')
+      httpResponse.on('data', chunk => rbody.push(chunk))
+      httpResponse.on('end', () => {
+        const recurlyRequest = new Request(options.method, options.path, requestBody)
+        const recurlyResponse = Response.build(httpResponse, rbody, recurlyRequest)
+        resolve(recurlyResponse)
+      })
+    })
+    httpRequest.on('error', reject)
+    if (requestBody) {
+      httpRequest.write(requestBody)
+    }
+    httpRequest.end()
+  })
+}
+
 class Response {
-  constructor (response, body, request) {
-    this.request = request
-    this.body = null
-    if (body.length > 0) {
-      this.body = JSON.parse(body.join(''))
+  static build (response, body, request) {
```

Comment:
I don't love this since the reason to not make it a constructor is for easier mocking. I certainly empathize though, since the only way around it would be to stub out much of what this method does. I'm just going to rest on it and say that this is probably the best way to make `Response` part of the mock.